src/components/Viewport.ts

```diff
@@ -445,6 +447,19 @@ export default class Viewport {
       const newCurrentIndex = Math.max(index - 1, panelManager.getRange().min);
       this.currentPanel = panelManager.get(newCurrentIndex);
     }
+
+    // Update checked indexes in infinite mode
+    if (deleteCount > 0) {
+      state.checkedIndexes.forEach((indexes, idx) => {
+        const [min, max] = indexes;
+        // Can fill part of indexes in range
+        if (isBetween(index, min, max)) {
+          // Remove checked index from list
+          state.checkedIndexes.splice(idx, 1);
+        }
```

Comment:
@WoodNeck

There are a lot of splice codes in forEach. I'm afraid we'll have an index problem, so please check.

Reply:
Sure, that must be fixed.

Reply:
I've checked it but this looks outdated, already fixed in previous commit.